R/SCE2AnnData.R

```diff
@@ -226,14 +224,28 @@ SCE2AnnData <- function(sce, X_name = NULL, assays = TRUE, colData = TRUE,
     adata$varp <- .convert_sce_pairs(sce, "rowPairs", "varp", rowPairs)
 
     if (!is.null(colnames(sce))) {
-        adata$obs_names <- colnames(sce)
+        if (is.null(adata$obs)) {
+            # hack to get an empty data frame with the right dimensions
+            adata$obs <- data.frame(i = rep(NA, ncol(sce)))[,-1]
+        }
+        # already convert to python because python DFs can have duplicates in
+        # their index
+        adata$obs <- r_to_py(adata$obs)
+        adata$obs$index <- colnames(sce)
```

Comment:
Is it better to set `index` here rather than `obs_names`?

Reply:
`obs_names` is not an argument of the `anndata.AnnData(...)` constructor, unfortunately.